nasl/nasl_snmp.c

```diff
@@ -599,12 +721,46 @@ nasl_snmpv3_get (lex_ctxt *lexic)
       else if (!strcasecmp (privproto, "aes"))
         pproto = 1;
       else
-        return array_from_snmp_result (-2, "privproto should be des or aes");
+        {
+          nasl_perror (lexic, "Unsupported SNMP function.");
+        }
+      return array_from_snmp_result (-2, "privproto should be des or aes");
     }
 
   g_snprintf (peername, sizeof (peername), "%s:%s:%d", proto,
               plug_get_host_ip_str (lexic->script_infos), port);
-  ret = snmpv3_get (peername, username, authpass, aproto, privpass, pproto,
-                    oid_str, &result);
+
+  if (action == NASL_SNMP_GET)
+    {
+      ret = snmpv3_get (peername, username, authpass, aproto, privpass, pproto,
+                        oid_str, &result);
+    }
+  else if (action == NASL_SNMP_GETNEXT)
+    {
+#ifdef HAVE_NETSNMP
+      ret = snmpv3_getnext (peername, username, authpass, aproto, privpass,
+                            pproto, oid_str, &result);
+#else
+      nasl_perror (lexic,
+                   "The SNMP function getnext is only available with libsnmp.");
+      return NULL;
+#endif /* HAVE_NETSNMP */
+    }
+  else
+    {
+      nasl_perror (lexic, "Unsupported SNMP function.");
```

Comment:
Seems this error logging was only added for SNMPv3. What about SNMPv1 and SNMPv2c?